### PR TITLE
Update csharp README and fix .NET 3.5 build error

### PR DESCRIPTION
**Pull request by JohnHBrock**

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by JohnHBrock** [comment]
@jskeet Can you take a look? Thanks!

File: csharp/README.md [hunk 1]
```diff
@@ -31,28 +31,22 @@ which only uses features from C# 3 and earlier.
...
-[dotnet-test-nunit](https://github.com/nunit/dotnet-test-nunit). 
-`dotnet-test-nunit` can also run tests for .NET 4.5+, so to run the tests 
-for both .NET Core and .NET 4.5, you can simply open the 
-`Package Manager Console` in Visual Studio and execute:
-```
-dotnet test Google.Protobuf.Test
-```
+The unit tests use [NUnit 3](https://github.com/nunit/nunit). NUnit doesn't yet
+support `dotnet test`, so for now the test project is a console application 
+using NUnitLite. Simply run `Google.Protobuf.Test.exe` to run the unit tests.
```
Comment (jskeet) [inline csharp/README.md]: I'd add "either directly, or using `dotnet run`".

File: csharp/README.md [hunk 2]
```diff
@@ -62,35 +56,15 @@ to make enabling .NET 3.5 support relatively painless in case you require it.
...
-([possibly related issue](https://github.com/nunit/nunit/issues/1480)).
-
-If you still want to run the .NET 4.5 and .NET Core tests, you can do so by 
-specifying the framework when using `dotnet-test-nunit`, i.e. from 
-`Package Manager Console` in Visual Studio:
-```
-dotnet test Google.Protobuf.Test --framework netcoreapp1.0
-dotnet test Google.Protobuf.Test --framework net451
-```
+To enable .NET 3.5 support, you must edit the `TargetFramework` elements of
```
Comment (jskeet) [inline csharp/README.md]: `TargetFrameworks` (plural)

File: csharp/README.md [hunk 3]
```diff
@@ -62,35 +56,15 @@ to make enabling .NET 3.5 support relatively painless in case you require it.
...
-dotnet test Google.Protobuf.Test --framework netcoreapp1.0
-dotnet test Google.Protobuf.Test --framework net451
-```
+To enable .NET 3.5 support, you must edit the `TargetFramework` elements of 
+[src/Google.Protobuf/Google.Protobuf.csproj](src/Google.Protobuf/Google.Protobuf.csproj) 
+(and [src/Google.Protobuf.Test/Google.Protobuf.Test.csproj](src/Google.Protobuf.Test/Google.Protobuf.Test.csproj) 
+if you want to run the unit tests): 
+
+Open the .csproj file in a text editor and simply add `net35` to the list of 
+target frameworks, noting that the `TargetFramework` element appears twice in
```
Comment (jskeet) [inline csharp/README.md]: `TargetFrameworks` again

File: csharp/src/Google.Protobuf.Test/Program.cs [hunk 4]
```diff
@@ -38,7 +38,11 @@ class Program
     {
         public static int Main(string[] args)
         {
-            return new AutoRun(typeof(Program).GetTypeInfo().Assembly).Execute(args);
+            #if NET35
+                return new AutoRun(typeof(Program).Assembly).Execute(args);
```
Comment (jskeet) [inline csharp/src/Google.Protobuf.Test/Program.cs]: Out of interest, have you built and run this?

Comment (JohnHBrock) [inline csharp/src/Google.Protobuf.Test/Program.cs]: Yes, it seems to work ok.

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

File: csharp/README.md (hunk 1 above)
Comment (JohnHBrock) [inline csharp/README.md]: Done.

File: csharp/README.md (hunk 2 above)
Comment (JohnHBrock) [inline csharp/README.md]: Fixed.

File: csharp/README.md (hunk 3 above)
Comment (JohnHBrock) [inline csharp/README.md]: Fixed.

**Comment by jskeet** [comment]
Thanks - LGTM; will merge when green, modulo me being awake.

**Pull request merged**